content/en/docs/Reference/kairos-factory.md

```diff
@@ -17,75 +17,357 @@ For the newcomer or someone who simply needs an immutable OS with k3s and edgeVP
+| -f           | Sets the framework version to use                        | Any valid framework version | latest          |
+| --fips       | Use FIPS framework for FIPS 140-2 compliance images      | true,false                  | false           |
+| --k8sversion | Sets the provider version to use                         | Any valid provider version  | latest          |
+| --validate   | Sets the run to validate which only validates the system | true,false                  | true            |
+
+
+You can provide a generic Dockerfile that gets all this values and passes them down into kairos-init like we do under Kairos:
+
+```Dockerfile
+ARG BASE_IMAGE=ubuntu:20.04
+
+FROM quay.io/kairos/kairos-init:v0.2.6 AS kairos-init
+
+FROM ${BASE_IMAGE} AS base-kairos
+ARG VARIANT=core
+ARG MODEL=generic
+ARG TRUSTED_BOOT=false
+ARG KUBERNETES_DISTRO=k3s
+ARG KUBERNETES_VERSION=latest
+ARG VERSION
+
+COPY --from=kairos-init /kairos-init /kairos-init
+RUN /kairos-init -l debug -s install -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
+RUN /kairos-init -l debug -s init -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
+RUN /kairos-init -l debug --validate -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
+RUN rm /kairos-init
+```
+
+{{% alert title="Variant" color="note" %}}
+
+When selecting the `standard` variant, the produced image will contain the default kubernetes provider (k3s) with the latest published version of k3s, the Kairos provider for kubernetes and some extra k8s utils like k9s or kube-vip.
+
+If you want to override those see the flags `-k` and `--k8sversion`
+
+{{% /alert %}}
+
+## Phases
+
+kairos-init is divided in 2 phases, one its the install phase which install all nedeed packages and binaries and the other is the init phase, which gets the system ready. This are the main parts of each phase:
+
+Install:
+ - Install required packages via system package manager
+ - Install [kairos-framework]() into the system
+ - Install k8s provider and tools if its an standard image
+
+Init:
+ - Fill the /etc/kairos-release data (needed for upgrades, grub booting, kernel cmdline, etc...)
+ - Remove Old kernel links and duplicates
+ - Get and link kernel to `/boot/vmlinuz`
+ - Remove all existing initrds
+ - Build a new initrd and link it to `/boot/initrd`
+ - Enable/disable needed services
+ - Run some workarounds (grub vs grub2 naming for example)
+ - Run system cleanup to avoid leftovers
+
+
+As you can see, both of this stages runs separatedly so you can hijack this in the middle and add or remove things.
+
+For example, its possible to add extra modules to be added to the initrd, or a specific kernel instead of the default latest one.
```

Comment:
```suggestion
For example, it's possible to add extra modules to be added to the initrd, or a specific kernel instead of the default latest one.
```